static compile broken

#### Describe the bug
-- Submodule update
Using the current master and doing make clean && make vw_static results in:

CMake Error at vowpalwabbit/CMakeLists.txt:31 (target_link_libraries):
  Object library target "vw_io" may not link to anything.

-- Configuring incomplete, errors occurred!
See also "/home/jl/programs/vowpal_wabbit/build/CMakeFiles/CMakeOutput.log".
See also "/home/jl/programs/vowpal_wabbit/build/CMakeFiles/CMakeError.log".
Makefile:22: recipe for target 'ensure_cmake_static' failed
make: *** [ensure_cmake_static] Error 1

#### To Reproduce
This is on: cat /proc/version
Linux version 4.15.0-117-generic (buildd@lcy01-amd64-008) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #118-Ubuntu SMP Fri Sep 4 20:02:41 UTC 2020

--- a/vowpalwabbit/CMakeLists.txt
+++ b/vowpalwabbit/CMakeLists.txt
@@ -17,18 +17,8 @@ else()
   target_compile_options(allreduce PUBLIC ${linux_flags})
 endif()
 
-# Prior to CMake 3.11 add_library requires at least 1 source file, so dummy.cc fulfills that need since we want to use target_sources.
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/dummy.cc "")
-add_library(vw_io OBJECT ${CMAKE_CURRENT_BINARY_DIR}/dummy.cc)
-target_sources(vw_io
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/io/io_adapter.h>
-    $<INSTALL_INTERFACE:io/io_adapter.h>
-  PRIVATE
-    io/io_adapter.cc
-)
-# This can't be correctly marked as PRIVATE until until CMake 3.12, until then it must be PUBLIC so that consumers link to it too.
-target_link_libraries(vw_io PUBLIC ZLIB::ZLIB)
+add_library(vw_io STATIC io/io_adapter.h io/io_adapter.cc)
+target_link_libraries(vw_io PRIVATE ZLIB::ZLIB)
 add_library(VowpalWabbit::io ALIAS vw_io)
 
 set(vw_all_headers

